modules/fedimint-lnv2-client/src/api.rs

```diff
@@ -129,6 +130,10 @@ where
             .cloned()
             .collect::<Vec<SafeUrl>>();
 
+        // Shuffling the gateways ensures that payments are distributed over the
+        // gateways evenly.
+        union.shuffle(&mut rand::thread_rng());
```

Comment:
This is useful because the `sort` called next is stable and we assume that most of the times guardians agree about the proposed gateways?

Reply:
Exactly - They will be ordered by number of votes but gateways with the same number of votes are in random order.